syntax = "proto3";

package egel_rpc;

service EgelRpc {
    rpc EgelCall(EgelText) returns (EgelResult) {}
    rpc EgelDependencies(EgelTexts) returns (EgelText) {}
    rpc EgelImport(EgelText) returns (EgelText) {}
    rpc EgelNodeInfo(EgelText) returns (EgelText) {}
}

message EgelText {
    string text = 1;
}

message EgelResult {
    string text = 1;
    bool exception = 2;
}

message EgelTexts {
    repeated string texts = 1;
}
